### PR TITLE
Clean up execute_in_process, target graph more directly for GraphDefinition.execute_in_process

### PR DESCRIPTION
**Pull request by dpeng817**
## Summary
There was a bunch of crufty implementation left over from the previous iteration of `execute_in_process`. This PR cleans most of that up. 
- got rid of `execute_in_process` standalone
- renamed core.execution.execute.py -> core.execution.execute_in_process.py
- run_config -> config on execute_in_process for graph.

In addition, when calling `graph.execute_in_process`, the called-upon graph now becomes the underlying graph for the ephemeral pipeline, which fixes the config inconsistency that was created.

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by alangenfeld — APPROVED** [review]

File: python_modules/dagster/dagster_tests/execution_tests/test_execute_in_process.py [hunk 1]
```diff
@@ -61,92 +43,6 @@ def emit_three():
     )
 
 
-def test_execute_solid_with_inputs():
```
Comment (sryza) [inline python_modules/dagster/dagster_tests/execution_tests/test_execute_in_process.py:64]: @dpeng817 it looks like we're losing a decent bit of test coverage here - is that intentional?

Comment (dpeng817) [reply python_modules/dagster/dagster_tests/execution_tests/test_execute_in_process.py:64]: I think any test coverage that we're losing is testing old, unused functionality of execute in process.

Comment (sryza) [reply python_modules/dagster/dagster_tests/execution_tests/test_execute_in_process.py:64]: It looks like `test_dynamic_output_solid` is exercising an `InProcessSolidResult.output_values` code path that isn't necessarily exercised elsewhere? Very possible I'm missing something.

Comment (dpeng817) [reply python_modules/dagster/dagster_tests/execution_tests/test_execute_in_process.py:64]: Hm I think you're right. I'll put up a PR to re-add a test around this.

**Pull request merged**